### ISSUE TITLE
Unable to configure Ingress + traffic splitting

### ISSUE THREAD

**Issue by GiuseppeChiesa-TomTom** [openservicemesh/osm#4769]
**Bug description**:
Following documentation I would like to configure [osm with traffic split between 2 services](https://release-v1-1.docs.openservicemesh.io/docs/demos/canary_rollout/), and have the root service be [exposed via ingress controller](https://release-v1-1.docs.openservicemesh.io/docs/demos/ingress_k8s_nginx/) (nginx).

Documentation reports how to test the traffic split only from a pod part of the mesh, but the ingress as per documentation should not have envoy injection, hence it's not possible to have the traffic split feature driven by the SMI TrafficSplit. 

**Affected area (please mark with X where applicable)**:
- Install                    [ ]
- SMI Traffic Access Policy  [ ]
- SMI Traffic Specs Policy   [ ]
- SMI Traffic Split Policy   [X]
- Permissive Traffic Policy  [X]
- Ingress                    [X]
- Egress                     [ ]
- Envoy Control Plane        [ ]
- CLI Tool                   [ ]
- Metrics                    [ ]
- Certificate Management     [ ]
- Sidecar Injection          [ ]
- Logging                    [ ]
- Debugging                  [ ]
- Tests                      [ ]
- Demo                       [ ]
- CI System                  [ ]

**Expected behavior**:
Ingress propagates the traffic honoring the traffic split rules defined in SMI TrafficSplit.

**Steps to reproduce the bug (as precisely as possible)**:
Follow the 2 guides to implement traffic split and integrate NGINX controller. The Ingress resource has to be configured with the root service implemented for the traffic split. 

**How was OSM installed?**:
Azure AKS Add-on.

**Anything else we need to know?**:

**Bug report archive**:

**Environment**:
- OSM version (use `osm version`):
```
CLI Version: version.Info{Version:"v1.1.1", GitCommit:"407bbedd5edb6ff9f1f51a4cabb95bedeb567312", BuildDate:"2022-05-10-15:55"}

MESH NAME   MESH NAMESPACE   VERSION   GIT COMMIT                                 BUILD DATE
osm         kube-system      v1.0.0    26886c4f16ae5cc92ddfe4c62c87cd82f1cd965c   2022-02-02-00:04
```
- Kubernetes version (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.0", GitCommit:"c2b5237ccd9c0f1d600d3072634ca66cefdf272f", GitTreeState:"clean", BuildDate:"2021-08-04T17:56:19Z", GoVersion:"go1.16.6", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.6", GitCommit:"07959215dd83b4ae6317b33c824f845abd578642", GitTreeState:"clean", BuildDate:"2022-03-30T18:28:25Z", GoVersion:"go1.16.12", Compiler:"gc", Platform:"linux/amd64"}
```
- Size of cluster (number of worker nodes in the cluster): 3 Nodes
- Others:

**Comment by shashankram** [comment]
Hi @GiuseppeChiesa-TomTom SMI traffic splitting is applicable only to pods participating in the service mesh. In OSM, we recommend not adding a sidecar to the ingress controller/gateway pod, and as such, it is technically not a part of the meshed pods managed by OSM controller so SMI traffic policies cannot be applied to un-meshed (pods without the proxy sidecar) ingress controller/gateway pods.

If your use case is to use SMI traffic splitting with ingress, then you could add the ingress pods to the mesh. In this case, you will not be able to use k8s Ingress to route traffic from your ingress gateway to backend pods, and instead would need to use SMI traffic policies. You will also need to configure inbound IP/Port exclusions so that traffic originating from external sources are able to access the ingress gateway pod, which would otherwise deny all traffic by default and only support mTLS based connections from downstream clients.

To summarize the above:
1. You cannot use SMI traffic splitting from ingress to backend pods when the ingress gateway is not managed by OSM
2. To use SMI traffic splitting with ingress, you need to make the ingress gateway a part of the mesh, and use SMI policies instead of the Ingress API to configure proxying from ingress->backend pods.

**Comment by shashankram** [comment]
> thanks for the clarification.
> 
> I think it would be interesting, perhaps, to consider a feature request to implement the point 2 via annotation. Similarly to the way you annotate an ingress for a pod to have the ingress controller discover & configure the routes, a TrafficSplit might be annotated to have osm create the proper additional SMI resource to make it reachable from the meshed ingress.

I am unsure about what you are suggesting here.

Since you are looking to use SMI traffic splitting for your ingress gateway, the only way this is possible is by having a sidecar on your ingress gateway and using SMI traffic policies instead of k8s Ingress routes to route traffic from the ingress gateway to your backend services.

**Comment by shashankram** [comment]
> @GiuseppeChiesa-TomTom Are you aware of [Gateway API](https://gateway-api.sigs.k8s.io/)? If you use a gateway-api compliant ingress controller (e.g Contour, NGINX, [etc](https://gateway-api.sigs.k8s.io/implementations/#implementation-status)), I believe you can accomplish the kind of [traffic splitting](https://gateway-api.sigs.k8s.io/v1alpha2/guides/traffic-splitting/) you're looking for on ingress.

Since @GiuseppeChiesa-TomTom mentioned using Nginx ingress, using the Gateway API is unlikely to be feasible as it is still being scoped by the k8s Nginx project (https://github.com/kubernetes/ingress-nginx/issues/7517).

**Comment by SomaSharathSurya** [comment]
> Hi @GiuseppeChiesa-TomTom SMI traffic splitting is applicable only to pods participating in the service mesh. In OSM, we recommend not adding a sidecar to the ingress controller/gateway pod, and as such, it is technically not a part of the meshed pods managed by OSM controller so SMI traffic policies cannot be applied to un-meshed (pods without the proxy sidecar) ingress controller/gateway pods.
> 
> If your use case is to use SMI traffic splitting with ingress, then you could add the ingress pods to the mesh. In this case, you will not be able to use k8s Ingress to route traffic from your ingress gateway to backend pods, and instead would need to use SMI traffic policies. You will also need to configure inbound IP/Port exclusions so that traffic originating from external sources are able to access the ingress gateway pod, which would otherwise deny all traffic by default and only support mTLS based connections from downstream clients.
> 
> To summarize the above:
> 
> 1. You cannot use SMI traffic splitting from ingress to backend pods when the ingress gateway is not managed by OSM
> 2. To use SMI traffic splitting with ingress, you need to make the ingress gateway a part of the mesh, and use SMI policies instead of the Ingress API to configure proxying from ingress->backend pods.


@shashankram Thanks for clarifying this. Would you be able to provide some examples on how to configure the option 2 ? Any idea if this is recommended for production workloads or are there limitations around it (as the [documentation](https://release-v1-1.docs.openservicemesh.io/docs/demos/ingress_k8s_nginx/) states that the ingress pods should not have envoy injection)
